### ISSUE TITLE
A bug in function generate_sub_keys

### ISSUE THREAD

**Issue by xiekch** [tarequeh/DES#9]
Hi, the code is well-written but I find a bug in the function `generate_sub_keys`. You don't initialize the `key_sets` before using it neither before calling the function. Thus the generated keys go wrong in my test.
My suggestion is as follows. Just add `memset(key_sets, 0, 17 * sizeof(key_set);` in the top of the function.

**Comment by xiekch** [comment]
You initialized the first subkey but it is not enough

**Comment by CreazyMouse** [comment]
yes, befer generate_sub_keys should add memset(key_sets, 0, 17 * sizeof(key_set));

**Comment by HelloTodayWolrd** [comment]
yes, thank you

**Comment by 123yongming** [comment]
Bravo!
